frontends/concrete-python/concrete/fhe/compilation/configuration.py

```diff
@@ -418,6 +418,204 @@ def _promotions(
+    # - tlu(pack(x, y)) :: 11-bits -> 8-bits
+
+    THREE_TLU_CASTED = "three-tlu-casted"
+    # ---------------------------------
+    # conditions:
+    # - (x.bit_width + y.bit_width) <= MAXIMUM_TLU_BIT_WIDTH
+    #
+    # bit-width assignment:
+    # - x :: 3-bits
+    # - y :: 8-bits
+    #
+    # execution:
+    # - x = tlu(x) :: 3-bits -> 11-bits
+    # - y = tlu(y) :: 8-bits -> 11-bits
+    # - tlu(pack(x, y)) :: 11-bits -> 8-bits
+
+    TWO_TLU_BIGGER_PROMOTED_SMALLER_CASTED = "two-tlu-bigger-promoted-smaller-casted"
+    # -----------------------------------------------------------------------------
+    # conditions:
+    # - (x.bit_width + y.bit_width) <= MAXIMUM_TLU_BIT_WIDTH
+    #
+    # bit-width assignment:
+    # - x :: 3-bits
+    # - y :: 8-bits -> 11-bits
+    #
+    # execution:
+    # - x = tlu(x) :: 3-bits -> 11-bits
+    # - tlu(pack(x, y)) :: 11-bits -> 8-bits
+
+    TWO_TLU_BIGGER_CASTED_SMALLER_PROMOTED = "two-tlu-bigger-casted-smaller-promoted"
+    # -----------------------------------------------------------------------------
+    # conditions:
+    # - (x.bit_width + y.bit_width) <= MAXIMUM_TLU_BIT_WIDTH
+    #
+    # bit-width assignment:
+    # - x :: 3-bits -> 11-bits
+    # - y :: 8-bits
+    #
+    # execution:
+    # - y = tlu(y) :: 8-bits -> 11-bits
+    # - tlu(pack(x, y)) :: 11-bits -> 8-bits
+
+    CHUNKED = "chunked"
+    # ---------------
+    # bit-width assignment:
+    # - x :: 3-bits
+    # - y :: 8-bits
+    #
+    # execution:
+    # - at least 4 TLUs
+    # - at most 9 TLUs
+    # - it's complicated...
+
+    @classmethod
+    def parse(cls, string: str) -> "BitwiseStrategy":
+        """
+        Convert a string to a BitwiseStrategy.
+        """
+
+        if isinstance(string, cls):
+            return string
+
+        if not isinstance(string, str):
+            message = f"{string} cannot be parsed to a {cls.__name__}"
+            raise TypeError(message)
+
+        string = string.lower()
+        for value in BitwiseStrategy:
+            if string == value.value:
+                return value  # pragma: no cover
+
+        message = (
+            f"'{string}' is not a valid '{friendly_type_format(cls)}' ("
+            f"{', '.join(v.value for v in BitwiseStrategy)})"
+        )
+        raise ValueError(message)
+
+    def can_be_used(self, x: ValueDescription, y: ValueDescription) -> bool:
+        """
+        Get if the strategy can be used for the bitwise operation.
+
+        Args:
+            x (ValueDescription):
+                description of the lhs of the bitwise operation
+
+            y (ValueDescription):
+                description of the rhs of the bitwise operation
+
+        Returns:
+            bool:
+                whether the strategy can be used for the bitwise operation
+        """
+
+        assert isinstance(x.dtype, Integer)
+        assert isinstance(y.dtype, Integer)
+
+        if self in {
+            ComparisonStrategy.ONE_TLU_PROMOTED,
```

Comment:
I don't get it. Why ComparisonStrategy vs BitwiseStrategy ?

Reply:
Fixed.